### DISCUSSION TITLE
(Via Discord) Expanding Jail System

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#7865] [Enhancement]
Source: `Chief5#2647` <https://discord.com/channels/452792793631555594/1054579120861876246/1113918628547547219>

**__CMI Suggestion:__**

**More Intricate Jailing**

*▹* **Bond System** *(toggleable true/false as not everyone would use it | pay money to release a player)*
>     ▴ Add a command to set a players bond amount, remove a players bond, and set bond multiplier *(configurable in settings)* 
>     ▴ If player has been jailed before, they could have a configured `Bond Multiplier` that increases their bond amount automatically from the bond command
>     ▴ GUI to display players bond amounts

*▹* **Workers System** *(toggleable true/false as not everyone would use it | work to earn freedom)*
>     ▴ If toggled true - players would be allowed to leave jail cell to complete *"tasks"*
>     ▴ Completed tasks would decrease jail time or bond amount *(configurable*)
>     ▴ Tasks could be defined in settings *(something like `Task Name: Action: break_block Material: STONE Amount: 64 DecreaseTime-Bond: true Time: 5m Bond: 100.0` etc.)*

*▹* **Probation System** *(toggleable true/false as not everyone would use it | configureable check-in times for players)*
>     ▴ Once players are unjailed, they will be placed on *probation*
>     ▴ Probation officers can be set as NPC's that players can come "check-in" to. Check-ins would be a configured schedule based on amount of times jailed, or the bond multipler.
>     ▴ If the player doesn't check-in within the configured time, plus grace period, they get rejailed or cuffed until they check-in

**Comment by mrfloris** [comment]
On a more personal level. I wouldn't mind if there was a way to make it more clear to players why they got jailed, especially if more than one person shares the same jail sell. A placeholder on a sign doesn't really give enough space. And it's super hard to control what shows up in a book. And then to figure out permissions for someone to read a personalized book in a lec.. I'm rambling. 

I am hoping for some feature tweaking that allows for more 'gameplay' jail, besides 'server management' jail. So we can also use it for event or pvp fights, arenas, RPG stuff, MMO stuff, and not just for 'swearing', or 'grief'. A who what why that's easier to understand and figure out, and a way for the jailed player to resolve their situation even if there's no staff online .. would be nice. 

I don't know yet how I want that to be, but I think it's worth a discussion.

**Reply by Chiefink5** [reply]
ohh that would be INSANE!! i would love to see something like this. without a doubt!

**Comment by MC-MrBirdy** [comment]
Interesting ideas for an expanded jail system.
Just some ideas that came to my mind when reading your suggestions:
- For the tasks: "the option to define a region per task"
- For the bonds: "beside money allow items (with nbt) to be setup"
- For the probation: "like the ic command allow blocks and entities"

As for the making it clear to players why they are jailed. Are holograms no option for when more space is needed?
( I personally don't use signs/holograms/books, as the staff on my server sends an in-game mail with the reason/details. )

**Reply by Chiefink5** [reply]
great additions, i'd say!

as far as the holograms go, they can be used, but they can also be pretty over-bearing in some cases.. on of my jails had 24 cells and to display everyones reason, time, etc. would be a lot of holograms in a small area, or a few large ones that are out of the view of the cells in a sense. so i mean they work but theres definetly a better way to display everything, 

_(spacing this out because this could be another suggestion if its a good enough idea)_

which in my opinion would be in a GUI.-

the GUI could display configured heads for defined jails, that when clicked on would open another gui showing the cells for that jail, then you can click on a specific cell that would _then open another GUI_ showing the players in that cell. each player head would display their reason, their time, bond amount, if theyre allowed to be a worker, etc. with options for admins to take actions from there like
- release them from jail on probation
- increase/decrease bond
- increase/decrease time
- etc.
